crates/ruff_linter/src/rules/ruff/rules/abstract_method_in_normal_class.rs

```diff
@@ -0,0 +1,102 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{Stmt, StmtClassDef, StmtFunctionDef};
+use ruff_python_semantic::analyze::visibility::{abstract_decorator, AbstractDecoratorKind};
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for methods decorated with at least one of `abc`'s
+/// `abstractmethod`, `abstractclassmethod`, `abstractstaticmethod` and `abstractproperty`
+/// but defined within a normal class's body.
+///
+/// ## Why is this bad?
+/// Such abstract method decorators are only effective in an abstract class.
```

Comment:
Could we expand a bit more here with an example so the reader understands what will happen when the class is not an abstract class? 

The abstract method decorators prevent users from instantiating abstract classes, or inheriting from abstract classes without implementing all abstract methods by throwing an exception:
> TypeError: Can't instantiate abstract class [class] with abstract method [method]

For classes that inherit from multiple classes, for example a single base class and a mixin, it's _not_ enough that `abc.ABC` is the meta class for only one of them to achieve the behaviour above., even though `abc.ABC` will be part of the MRO of the subclass.

Example code:

```python
from abc import ABC, abstractmethod


class Base(ABC):
    @abstractmethod
    def hello(self) -> None:
        ...

    def __repr__(self) -> str:
        return f"message='{self.msg}'"


class Mixin: # should be: `Mixin(ABC)`:
    @abstractmethod
    def world(self) -> None:
        self.msg += " goodbye"


class FooBar(Mixin, Base):
    def __init__(self):
        self.msg = ""

    def hello(self) -> None:
        self.msg += "hello"

    # without `Mixin(ABC)`, omitting this does not raise an exception
    # def world(self) -> None:
    #     self.msg += " world"


# `abc.ABC` is part of the MRO
print(FooBar.mro()) # [<class '__main__.FooBar'>, <class '__main__.Mixin'>, <class '__main__.Base'>, <class 'abc.ABC'>, <class 'object'>] 

fb = FooBar()
fb.hello()
fb.world()
print(str(fb))  # message='hello goodbye'
```